core/src/test/java/org/elasticsearch/index/seqno/CheckpointsIT.java

```diff
@@ -61,16 +61,19 @@ public void testCheckpointsAdvance() throws Exception {
                     XContentHelper.toString(shardStats.getSeqNoStats(),
                         new ToXContent.MapParams(Collections.singletonMap("pretty", "false"))));
                 final Matcher<Long> localCheckpointRule;
+                final Matcher<Long> globalCheckpointRule;
                 if (shardStats.getShardRouting().primary()) {
                     localCheckpointRule = equalTo(numDocs - 1L);
+                    globalCheckpointRule = equalTo(numDocs - 1L);
```

Comment:
what happens when indexing completes, relocation happens and then (and only then) the global updater kicks in. I *think* we still have an issue, if true, let's just relax global checkpoint for now. I'm good with doing all of that as a separate fix in the interest of getting this in.